tools/mascot/src/main/java/dk/alexandra/fresco/tools/mascot/MascotResourcePoolImpl.java

```diff
@@ -0,0 +1,137 @@
+package dk.alexandra.fresco.tools.mascot;
+
+import dk.alexandra.fresco.commitment.HashBasedCommitment;
+import dk.alexandra.fresco.commitment.HashBasedCommitmentSerializer;
+import dk.alexandra.fresco.framework.network.Network;
+import dk.alexandra.fresco.framework.network.serializers.ByteSerializer;
+import dk.alexandra.fresco.framework.network.serializers.StrictBitVectorSerializer;
+import dk.alexandra.fresco.framework.sce.resources.ResourcePoolImpl;
+import dk.alexandra.fresco.framework.util.Drbg;
+import dk.alexandra.fresco.framework.util.ExceptionConverter;
+import dk.alexandra.fresco.framework.util.StrictBitVector;
+import dk.alexandra.fresco.tools.cointossing.CoinTossing;
+import dk.alexandra.fresco.tools.mascot.field.FieldElementSerializer;
+import dk.alexandra.fresco.tools.mascot.prg.FieldElementPrg;
+import dk.alexandra.fresco.tools.mascot.prg.FieldElementPrgImpl;
+import dk.alexandra.fresco.tools.ot.base.RotBatch;
+import dk.alexandra.fresco.tools.ot.otextension.BristolRotBatch;
+import dk.alexandra.fresco.tools.ot.otextension.OtExtensionResourcePool;
+import dk.alexandra.fresco.tools.ot.otextension.OtExtensionResourcePoolImpl;
+import dk.alexandra.fresco.tools.ot.otextension.RotImpl;
+import dk.alexandra.fresco.tools.ot.otextension.RotList;
+import java.math.BigInteger;
+import java.security.MessageDigest;
+import java.util.Map;
+
+public class MascotResourcePoolImpl extends ResourcePoolImpl implements MascotResourcePool {
+
+  private final Map<Integer, RotList> seedOts;
+  private final int instanceId;
+  private final BigInteger modulus;
+  private final int modBitLength;
+  private final int lambdaSecurityParam;
+  private final int prgSeedLength;
+  private final int numCandidatesPerTriple;
+  private final FieldElementPrg localSampler;
+  private final FieldElementSerializer fieldElementSerializer;
+  private final StrictBitVectorSerializer strictBitVectorSerializer;
+  private final ByteSerializer<HashBasedCommitment> commitmentSerializer;
+  private final MessageDigest messageDigest;
+
+  /**
+   * Creates {@link MascotResourcePoolImpl}.
+   */
+  public MascotResourcePoolImpl(int myId, int noOfParties,
```

Comment:
This constructor is kind of a mess. At the least we should document the parameters. Although, even then it's confusing with so many parameters (take a look at the calls to this constructor in TestMascotResourcePoolImpl and see if you can figure out what is being constructed). A possible solution could be a more readable strategy for constructing the resource pool like _Step Builder_ (https://devliving.online/stepbuilder-builder-that-guides-you-through-the-steps/). Or perhaps the MascotResoucePool is just handling to many parameters?

Reply:
Whereas I agree it is a mess, the proposed solution has the disadvantage of losing final fields.
One could create intermediary objects that holds related types of data,